devito/passes/clusters/aliases.py

```diff
@@ -922,10 +922,14 @@ def lower_schedule(schedule, meta, sregistry, ftemps):
 
         # Drop or weaken parallelism if necessary
         for d, v in meta.properties.items():
-            if any(i.is_Modulo for i in ispace.sub_iterators[d]):
-                properties[d] = normalize_properties(v, {SEQUENTIAL})
-            elif d not in writeto.itdims:
-                properties[d] = normalize_properties(v, {PARALLEL_IF_PVT}) - {ROUNDABLE}
+            try:
```

Comment:
why?

Reply:
This one is sneacky but when you define 2- dimensional IndexDerivatives (i.e `u(x+i*hy, y - i * hy)`) you end up here with `d = (x, y)` which isn't a valid key (you end up in the separable case below)

Reply:
Can you add a comment explaining this right before `pass` inside the `except KeyError` branch?